### PR TITLE
Add temporary logs on allocator

### PR DESCRIPTION
**Pull request by kamaradclimber**
We want to see what happens exactly for roles with a quota.
This patch is likely to induce conflicts when we'll rebase on later
versions

Change-Id: Ic5850eea4679eea73f12794371c3285663ac5bdb

### CODE REVIEW

File: src/master/allocator/mesos/hierarchical.cpp [hunk 1]
```diff
@@ -2447,6 +2459,8 @@ bool HierarchicalAllocatorProcess::allocatable(
       ? &options.minAllocatableResources.get()
       : nullptr;
 
+
```
Comment (oaitmous) [inline src/master/allocator/mesos/hierarchical.cpp]: remove these empty lines

File: src/master/allocator/mesos/hierarchical.cpp [hunk 2]
```diff
@@ -1847,12 +1853,14 @@ void HierarchicalAllocatorProcess::__allocate()
...
       foreach (const string& frameworkId_, frameworkSorter->sort()) {
         Resources available = slave.getAvailable();
+        VLOG(2) << "[CRITEO] slave " << slaveId << " has following available resources: " << available;
 
         // Offer a shared resource only if it has not been offered in this
         // offer cycle to a framework.
         available -= offeredSharedResources.get(slaveId).getOrElse(Resources());
 
         if (available.allocatableTo(role).empty()) {
+          VLOG(2) << "[CRITEO] " + role + " cannot be allocated any resources on slave " << slaveId << " because of allocatableTo method";
```
Comment (oaitmous) [inline src/master/allocator/mesos/hierarchical.cpp]: ...**due to the** allocatableTo method

File: src/master/allocator/mesos/hierarchical.cpp [hunk 3]
```diff
@@ -1924,6 +1934,7 @@ void HierarchicalAllocatorProcess::__allocate()
         // a reservation. Otherwise, this role is not going to get any
         // allocation. We can safely `continue` here.
         if (toAllocate.empty()) {
+          VLOG(2) << "[CRITEO] " + role + " has no reserved resources on that slave and there is no unreserved resources on that slave " << slaveId;
```
Comment (oaitmous) [inline src/master/allocator/mesos/hierarchical.cpp:1937]: has no reserved resource on the slave, which has no unreserved resources left

File: src/master/allocator/mesos/hierarchical.cpp [hunk 4]
```diff
@@ -1823,6 +1823,10 @@ void HierarchicalAllocatorProcess::__allocate()
     CHECK(slaves.contains(slaveId));
     Slave& slave = slaves.at(slaveId);
 
+    VLOG(2) << "[Criteo] Here is the SORTED list of roles that will be considered to get resources for slave " << slaveId;
```
Comment (oaitmous) [inline src/master/allocator/mesos/hierarchical.cpp]: lowercase for **sorted**

**Comment by kamaradclimber** [comment]
all done

**Pull request merged**